### PR TITLE
[RN] Add a timeout for loading the configuration

### PR DESCRIPTION
**Pull request by saghul**
I chose 8 seconds pretty arbitrarily, let's pick a value which seems reasonable.

Tip for testing: use the builtin network conditioner in iOS and set packet-loss to 100%.

### CODE REVIEW

**Comment by lyubomir** [comment]
@paweldomas, could you please review this PR (on Monday)? You discovered the need for a timeout last Thursday.

What immediately jumps to me during my first read of this PR is:
* You mentioned that react-native's fetch implementation didn't respect an input timeout but the Internet was talking about workarounds. I didn't see a comment on the workarounds.
* The fetch is internally an XHR and we're tracking the XHRs in NetworkActivityIndicator so that we may display a (spinning) progress indicator while there's at least one XHR in progress. The timeout implemented by this PR seems to me to be leaving its XHR in progress (as far as NetworkActivityIndicator in concerned) though to our app the XHR is "done".

**Comment by paweldomas** [comment]
> What immediately jumps to me during my first read of this PR is:

>>You mentioned that react-native's fetch implementation didn't respect an input timeout but the Internet was talking about workarounds. I didn't see a comment on the workarounds.

What Saul did was the workaround from the Internet. I'm not sure what you mean by "comment". Do you mean that it should explain why are we using timeout and not XHR.timeout ?

**Comment by paweldomas** [comment]
I wonder why are we even tracking those network requests in the first place ? We don't need that to display the spinner, because we know when we started to join the conference and when was it joined. I don't think we should be using network-activity for anything other than debugging eventually.

File: react/features/base/lib-jitsi-meet/functions.js [hunk 1]
```diff
@@ -148,5 +150,5 @@ export function loadConfig(url: string) {
         return value;
     });
 
-    return promise;
+    return timeoutPromise(promise, timeoutMs);
```
Comment (lyubomir) [inline react/features/base/lib-jitsi-meet/functions.js]: Actually, the problem that we're trying to work around with the timeout is the network request. What's being timed out here is not only the network request - `loadScript` - but also the subsequent processing which we technically didn't really set out to time out.

**Pull request merged**